mtk.jl

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env julia
+###### AC-OPF using JuMP ######
+#
+# implementation reference: https://github.com/lanl-ansi/PowerModelsAnnex.jl/blob/master/src/model/ac-opf.jl
+# only the built-in AD library is supported
+#
+
+import PowerModels
+import Ipopt
+import ModelingToolkit, Optimization, OptimizationMOI
+import ModelingToolkit: ≲
+
+function solve_opf(file_name)
+    time_data_start = time()
+
+    data = PowerModels.parse_file(file_name)
+    PowerModels.standardize_cost_terms!(data, order=2)
+    PowerModels.calc_thermal_limits!(data)
+    ref = PowerModels.build_ref(data)[:it][:pm][:nw][0]
+
+    data_load_time = time() - time_data_start
+
+
+    time_model_start = time()
+    vars = []
+
+    ModelingToolkit.@variables va[keys(ref[:bus])]
+    ModelingToolkit.@variables vm[keys(ref[:bus])] = ones(length(ref[:bus])) (bounds = [(ref[:bus][i]["vmin"], ref[:bus][i]["vmax"]) for i in keys(ref[:bus])])
+    vars = vcat(vars, [va[i] for i in keys(ref[:bus])], [vm[i] for i in keys(ref[:bus])])
+    ModelingToolkit.@variables pg[keys(ref[:gen])] (bounds = [(ref[:gen][i]["pmin"], ref[:gen][i]["pmax"]) for i in keys(ref[:gen])])
+    ModelingToolkit.@variables qg[keys(ref[:gen])] (bounds = [(ref[:gen][i]["qmin"], ref[:gen][i]["qmax"]) for i in keys(ref[:gen])])
+    vars = vcat(vars, [pg[i] for i in keys(ref[:gen])], [qg[i] for i in keys(ref[:gen])])
+    i_inds, j_inds, l_inds = maximum(first.(ref[:arcs])), maximum(getindex.(ref[:arcs], Ref(2))), maximum(last.(ref[:arcs]))
+    ModelingToolkit.@variables p[1:i_inds, 1:j_inds, 1:l_inds]
+    ModelingToolkit.@variables q[1:i_inds, 1:j_inds, 1:l_inds]
+
+    for (l, i, j) in ref[:arcs]
+        push!(vars, p[l, i, j])
+        ModelingToolkit.setmetadata(p[l, i, j], ModelingToolkit.VariableBounds, (-ref[:branch][l]["rate_a"], ref[:branch][l]["rate_a"]))
+    end
+
+    loss = sum(gen["cost"][1] * pg[i]^2 + gen["cost"][2] * pg[i] + gen["cost"][3] for (i, gen) in ref[:gen])
+
+    cons = Array{Union{ModelingToolkit.Equation,ModelingToolkit.Inequality}}([])
+    for (i, bus) in ref[:ref_buses]
+        push!(cons, va[i] ~ 0)
+    end
+
+    for (i, bus) in ref[:bus]
+        bus_loads = [ref[:load][l] for l in ref[:bus_loads][i]]
+        bus_shunts = [ref[:shunt][s] for s in ref[:bus_shunts][i]]
+        push!(cons,
+            sum(p[a...] for a in ref[:bus_arcs][i]) ~
+                (isempty(ref[:bus_gens][i]) ? 0.0 : sum(pg[g] for g in ref[:bus_gens][i])) -
```

Comment:
```suggestion
                sum(pg[g] for g in ref[:bus_gens][i]; init = 0.0) -
```
Perhaps? I didn't try locally